mlir/lib/IR/Operation.cpp

```diff
@@ -245,8 +245,6 @@ void Operation::setOperands(ValueRange operands) {
 /// than the range pointed to by 'start'+'length'.
 void Operation::setOperands(unsigned start, unsigned length,
                             ValueRange operands) {
-  assert((start + length) <= getNumOperands() &&
-         "invalid operand range specified");
```

Comment:
Removing an assert, especially at the core doesn't look safe - can you state the reason for doing so? I tried looking at the instances where you've tried using it, but wasn't able to reason that out.